File: app/backend/models/items.py
from sqlalchemy import Column, Integer, String

from db import Base


class Item(Base):
    __tablename__ = "items"

    id = Column(Integer, primary_key=True)
    image = Column(String(255))
    title = Column(String(100))
    model = Column(String(100))
    description = Column(String(255))
    price = Column(Integer)
    storage = Column(Integer)
    chip = Column(String(100))
    display = Column(String(100))
    color = Column(String(100))


Car = Item

File: app/frontend/src/api/items.js
import { api } from './client'

function normalizeProduct(item) {
  return {
    id: item.id,
    name: item.title || item.model || 'Product',
    year: item.model || '',
    description: item.description || '',
    price: Number(item.price ?? 0),
    storage: item.storage,
    chip: item.chip,
    display: item.display,
    color: item.color,
    image: item.image || '',
    model: item.model || '',
    raw: item,
  }
}

export async function fetchProducts() {
  const items = await api.get('/items')
  return (Array.isArray(items) ? items : []).map(normalizeProduct)
}

export async function fetchProductById(id) {
  const item = await api.get(`/items/${id}`)
  return normalizeProduct(item)
}
